js/gestionPresupuesto.js

```diff
@@ -1,25 +1,114 @@
-// TODO: Crear las funciones, objetos y variables indicadas en el enunciado
+let presupuesto=0;
+let gastos=[];
+let idGasto=0;
 
-// TODO: Variable global
 
+function actualizarPresupuesto(presu) {
+    if (isNaN(presu) || presu <0){
+        return -1
+    }
 
-function actualizarPresupuesto() {
-    // TODO
+    presupuesto=presu
+    return presupuesto
 }
 
 function mostrarPresupuesto() {
-    // TODO
+    return `Tu presupuesto actual es de ${presupuesto} €`
 }
 
-function CrearGasto() {
-    // TODO
+function CrearGasto(descripcion, valor,fecha,...etiquetas) {
+
+    this.descripcion=descripcion
+    this.valor=(isNaN(valor) || valor<=0) ? 0 : valor
+    this.fecha=(!fecha || isNaN(Date.parse(fecha))) ? Date.now() : Date.parse(fecha);
+    this.etiquetas=(!etiquetas)? [] : etiquetas
```

Comment:
En lugar de pasar el listado de etiquetas directamente, puedes utilizar `this.anyadirEtiquetas`. Así evitas que las etiquetas se puedan crear duplicadas.